MetaMask/metamask-extension
# Make `eth_accounts` return all permitted accounts

## Explanation

It's high time we make `eth_accounts` return all permitted accounts instead of only the most recently selected one, just like the inventors of arrays intended. There are no associated UI changes. See #18515 for details.

Fixes #18515

## Manual Testing Steps

This change should not affect any functionality within the extension, but may cause unexpected breakages in dapps. We should conduct basic compatibility testing with our own e2e test dapp and the top X currently used dapps in MetaMask. We should consider shipping this change to Mobile at the same time.

## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
